.github/workflows/_buildpacks-prepare-release.yml

```
@@ -23,30 +33,40 @@ on:
 jobs:
   prepare-release:
     name: Prepare Release
-    runs-on: pub-hk-ubuntu-22.04-small
+    runs-on: ${{ inputs.ip_allowlisted_runner }}
     steps:
+      - name: Get token for GH application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ inputs.app_id }}
+          private_key: ${{ secrets.app_private_key }}
+
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          token: ${{ steps.generate-token.outputs.app_token }}
+
+      - name: Configure git
+        run: |
+          git config --global user.name ${{ env.BOT_USER }}
+          git config --global user.email ${{ env.BOT_EMAIL }}
+
+      - name: Install Languages CLI
+        uses: heroku/languages-github-actions/.github/actions/install-languages-cli@main
 
       - name: Bump versions and update changelogs
         id: prepare
-        uses: heroku/languages-github-actions/.github/actions/prepare-release@main
-        with:
-          bump: ${{ inputs.bump }}
-          declarations_starting_version: ${{ inputs.declarations_starting_version }}
+        run: |
+          ARGS=("--bump" "${{ inputs.bump }}" "--repository-url" "https://github.com/${{ github.repository }}")
+          if [ -n "${{ inputs.declarations_starting_version }}" ]; then
+            ARGS+=("--declarations-starting-version" "${{ inputs.declarations_starting_version }}")
+          fi
+          actions prepare-release "${ARGS[@]}"
```

Would something like this work, to avoid the need to generate the args list? 

This makes use of the `format()` function:
https://docs.github.com/en/actions/learn-github-actions/expressions#format

...plus their ternery-like feature:
https://docs.github.com/en/actions/learn-github-actions/expressions#example

```suggestion
        run: actions prepare-release --bump "${{ inputs.bump }}" --repository-url "https://github.com/${{ github.repository }}" ${{ inputs.declarations_starting_version && format('--declarations-starting-version {0}', inputs.declarations_starting_version) }}
```

---

hmm, would have to try that to see if it works.